### PR TITLE
fix mssql and r2dbc integration

### LINKED ISSUE #15032: Can't start the app after generating a project with MSSQL : Cannot insert explicit value when IDENTITY_INSERT is set to OFF
##### **Overview of the issue**

After generating a project with MSSQL, and starting the app with IntelliJ, we get the following error : 

```
t.j.c.liquibase.AsyncSpringLiquibase     : Liquibase could not start correctly, your database is NOT ready: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/00000000000000_initial_schema.xml::00000000000001::jhipster:
     Reason: liquibase.exception.DatabaseException: java.sql.BatchUpdateException: Cannot insert explicit value for identity column in table 'jhi_user' when IDENTITY_INSERT is set to OFF.
```

Even though in the `00000000000000_initial_schema.xml` we have the following properties :

```xml
        <ext:loadData
                  file="config/liquibase/data/user.csv"
                  separator=";"
                  identityInsertEnabled="true"
                  tableName="jhi_user">
            <column name="activated" type="boolean"/>
            <column name="created_date" type="timestamp"/>
        </ext:loadData>
```

With the `identityInsertEnabled`, set to `true` (generated like this by default), we still can't start the app properly because liquibase can't manage to load the custom data in the CSV.

##### **Motivation for or Use Case**

Liquibase does not work properly right after generating the project, and we did not add any custom code.

##### **Reproduce the error**

Using Java 15, generating a project with MSSQL (with the JHipster configuration given below), and then running the project with IntelliJ.

##### **JHipster Version(s)**

JHipster 7.0.1

##### **JHipster configuration**

```json
{
  "generator-jhipster": {
    "entitySuffix": "Entity",
    "dtoSuffix": "JhiDto",
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "myApp",
    "jhipsterVersion": "7.0.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "testFrameworks": ["cypress", "gatling", "cucumber"],
    "pages": [],
    "creationTimestamp": 1621254883770,
    "serviceDiscoveryType": false,
    "reactive": true,
    "authenticationType": "jwt",
    "packageName": "my.app",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "sql",
    "devDatabaseType": "mssql",
    "prodDatabaseType": "mssql",
    "buildTool": "maven",
    "serverSideOptions": ["enableSwaggerCodegen:true"],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": true,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": true,
    "nativeLanguage": "fr",
    "packageFolder": "my/app",
    "jwtSecretKey": "My JWT Secret Key",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["fr", "en"]
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

We do not have the generated `.jhipster` directory since it is still a blank project.

##### **Browsers and Operating System**

Ubuntu 20.04

- [ x ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by atomfrede**
This PR fixes all issues mentioned in #15032 

* using `asEntity('User')` everywhere
* correcting the mssql r2dbc url
* adding the necessary setting to liquibase migration

closes #15032

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs [hunk 1]
```diff
@@ -147,6 +152,11 @@ _%>
             <column name="activated" type="boolean"/>
             <column name="created_date" type="timestamp"/>
         </<% if (prodDatabaseType === 'mssql') { %>ext:<% } %>loadData>
+        <% if (prodDatabaseType === 'mssql') { %>
+        <sql dbms="mssql">
+            SET IDENTITY_INSERT jhi_user ON;
```
Comment (DanielFran) [inline generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs:157]:
```suggestion
            SET IDENTITY_INSERT jhi_user OFF;
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

File: generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs (hunk 1 above)
Comment (atomfrede) [reply generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs:157]: :facepalm: You are right.

**Comment by DanielFran** [comment]
@atomfrede @jhipster/developers We are using an explicit liquibase extension [sabomichal/liquibase-mssql](https://github.com/sabomichal/liquibase-mssql#change-loaddata) that allow defining the property _identityInsertEnabled="true"_ to _allows explicit values to be inserted into the identity column of a table_, so the same as defining "_SET IDENTITY_INSERT jhi_user ON/OFF_".
The use of this specific extension was due to no support for this MSSQL property in 2009: https://liquibase.jira.com/wiki/spaces/CONTRIB/pages/1998867/MS+SqlServer+Extensions

In the meantime, the support of official MSSQL is up-to-date, so we probably should migrate back to the [official liquibase extension](https://github.com/liquibase/liquibase-mssql).

So there are 2 possibilities:
1 - exclude the use of the property "[_identityInsertEnabled="true"_](https://github.com/jhipster/generator-jhipster/pull/15264/files#diff-ae745829fb5d7160e4baa8d1789c80a16a6f953f86aeec1720836a38b1d0d384R146)" explicitly for reactive and define "_SET IDENTITY_INSERT jhi_user ON/OFF_" only for reactive
2 - Drop use of sabomichal/liquibase-mssql in favour of the liquibase/liquibase-mssql and define "SET IDENTITY_INSERT jhi_user ON/OFF" like in the official extension for both reactive and non-reactive database

File: generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs [hunk 2]
```diff
@@ -134,6 +134,11 @@ _%>
                                  referencedTableName="<%= user.entityTableName %>"/>
     <%_ } _%>
     <%_ if (authenticationType !== 'oauth2' || (authenticationType === 'oauth2' && applicationType !== 'microservice')) { _%>
+        <% if (prodDatabaseType === 'mssql') { %>
+        <sql dbms="mssql">
+            SET IDENTITY_INSERT jhi_user ON;
```
Comment (DanielFran) [inline generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs:139]: Needs review

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by atomfrede** [comment]
I would say let's use the official extension.

**Comment by atomfrede** [comment]
But very strange it doesn't work with `identityInsertEnabled` property.

**Comment by DanielFran** [comment]
Liquibase version we are using is version 4 and actually the extension was compatible with version 3, not sure there is any issue with the version4.

Anyway Liquibase should handle now reactive database also, so I agree the better is to switch with liquibase official extension.

**Comment by atomfrede** [comment]
@DanielFran Done. Works fine (tested with reactive only though, but don't see why it should not work with classical jdbc).

**Comment by DanielFran** [comment]
@atomfrede Can you please review tests?

**Comment by atomfrede** [comment]
> @atomfrede Can you please review tests?

On it. Seems easy to fix.

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
@atomfrede LGTM

Maybe we need to delete https://www.jhipster.tech/tips/004_tip_using_ms_sql_server.html

**Comment by atomfrede** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/42719.

Regarding the tip, there is already a disclaimer that the tip is not required anymore. But maybe it is time to delete it now.

**Comment by pascalgrimaud** [comment]
@atomfrede : approved

**Pull request merged**